code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -286,3 +286,164 @@
 	victim.hallucination += 100
 	add_cooldown()
+
+// ***************************************
+// *********** Hunter's Mark
+// ***************************************
+/datum/action/xeno_action/activable/hunter_mark
+	name = "Hunter's Mark"
+	action_icon_state = "hunter_mark"
+	mechanics_text = "Psychically mark a creature you have line of sight to, allowing you to sense its direction, distance and location with Psychic Trace."
+	plasma_cost = 75
+	keybind_signal = COMSIG_XENOABILITY_HUNTER_MARK
+	cooldown_timer = 60 SECONDS
+	///Target of the Hunter's Hunter's Mark ability; referenced by the Psychic Trace ability.
+	var/mob/living/hunter_mark_target
+
+/datum/action/xeno_action/activable/hunter_mark/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return
+
+	var/mob/living/carbon/xenomorph/hunter/X = owner
+
+	if(!isliving(A))
+		if(!silent)
+			to_chat(X, "<span class='xenowarning'>We cannot psychically mark this target!</span>")
+		return FALSE
+
+	var/mob/living/mark_target = A
+
+	if(mark_target == hunter_mark_target)
+		if(!silent)
+			to_chat(X, "<span class='xenowarning'>This is already our target!</span>")
+		return FALSE
+
+	if(mark_target == X)
+		if(!silent)
+			to_chat(X, "<span class='xenowarning'>Why would we target ourselves?</span>")
+		return FALSE
+
+	if(!X.line_of_sight(mark_target)) //Need line of sight.
+		if(!silent)
+			to_chat(X, "<span class='xenowarning'>We require line of sight to mark them!</span>")
+		return FALSE
+
+	return TRUE
+
+
+/datum/action/xeno_action/activable/hunter_mark/on_cooldown_finish()
+	to_chat(owner, "<span class='xenowarning'><b>We are able to impose our psychic mark again.</b></span>")
+	owner.playsound_local(owner, 'sound/effects/xeno_newlarva.ogg', 25, 0, 1)
+	return ..()
+
+
+/datum/action/xeno_action/activable/hunter_mark/use_ability(atom/A)
+
+	var/mob/living/carbon/xenomorph/X = owner
+	var/mob/living/victim = A
+
+	X.face_atom(victim) //Face towards the target so we don't look silly
+
+	to_chat(X, "<span class='xenodanger'>We prepare to psychically mark [victim] as our quarry.</span>")
+
+	if(!do_after(X, HUNTER_MARK_WINDUP, TRUE, target, BUSY_ICON_HOSTILE)) //Slight wind up
+		return fail_activate()
+
+	if(!X.line_of_sight(victim)) //Need line of sight.
+		to_chat(X, "<span class='xenowarning'>We lost line of sight to the target!</span>")
+		return fail_activate()
+
+	if(hunter_mark_target) //If we have an existing target, remove the registration.
+		UnregisterSignal(hunter_mark_target, COMSIG_PARENT_PREQDELETED)
+
+	hunter_mark_target = victim //Set our target
+
+	var/datum/action/xeno_action/psychic_trace/psychic_trace_set_target = X.actions_by_path[/datum/action/xeno_action/psychic_trace] //Set Psychic Trace's target
+	psychic_trace_set_target.psychic_trace_target = hunter_mark_target
+
+	RegisterSignal(hunter_mark_target, COMSIG_PARENT_PREQDELETED, .proc/unset_target) //For var clean up
+
```

Comment:
this COMSIG_PARENT_PREQDELETED should be 
COMSIG_PARENT_QDELETING, as well as all the other signals used here